server/src/main/java/org/apache/iotdb/db/service/thrift/impl/ClientRPCServiceImpl.java

```diff
@@ -148,6 +149,11 @@ public class ClientRPCServiceImpl implements IClientRPCServiceWithHandler {
 
   private final ISchemaFetcher SCHEMA_FETCHER;
 
+  @FunctionalInterface
+  public interface SelectResult<T, R, U> {
+    public void apply(T t, R r, U u) throws IoTDBException, IOException;
+  }
```

Comment:
```suggestion
  @FunctionalInterface
  private static final SelectResult OLD_SELECT_RESULT = (resp, queryExecution, fetchSize) -> resp.setQueryDataSet(QueryDataSetUtils.convertTsBlockByFetchSize(queryExecution, fetchSize));

  private static final SelectResult SELECT_RESULT = (resp, queryExecution, fetchSize) -> resp.setQueryResult(QueryDataSetUtils.convertQueryResultByFetchSize(queryExecution, fetchSize));

  @FunctionalInterface
  public interface SelectResult {
    void apply(TSExecuteStatementResp resp, IQueryExecution queryExecution, int fetchSize) throws IoTDBException, IOException;
  }
```

Reply:
reuse `OLD_SELECT_RESULT ` and `SELECT_RESULT ` in the following methods instead of creating new one  each time that method is called.